lib/publiccloud/provider.pm

```diff
@@ -337,8 +389,23 @@ Destroys the current terraform deployment
 sub terraform_destroy {
     my ($self) = @_;
     record_info('INFO', 'Removing terraform plan...');
-    assert_script_run('cd ' . TERRAFORM_DIR);
-    script_run('terraform destroy -no-color -auto-approve', TERRAFORM_TIMEOUT);
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        assert_script_run('cd ' . TERRAFORM_DIR . '/' . $self->conv_openqa_tf_name);
+    }
+    else {
+        assert_script_run('cd ' . TERRAFORM_DIR);
+    }
+    assert_script_run('terraform plan -no-color -out mydestroy', TERRAFORM_TIMEOUT);
+    my $ret = script_run('terraform apply -no-color -auto-approve mydestroy', TERRAFORM_TIMEOUT);
```

Comment:
Don't understand the `apply` here?!

Reply:
You have two way of doing the destroy: with `terraform destroy` command or with a `terraform plan -destroy` and then an `apply`. During my test I saw that I had less destroy issue on Azure with the `plan -destroy ; apply` variation. I don't know why but that's the case. We have several sporadic issues with Azure on destroy. With this method we can have less issues.

**PS:** I just show that the code on my lab and the one I pushed is not exactly the same here, as the `-destroy` option is missing in the `plan`... I will fix this.

Reply:
The missing `-destroy` explain it, thx. 

Do you need to apply the same variables/values as you did with the `terraform plan` before? Or is this only working, because you set the workspace and now terraform knows which statefile to use? (Just thinking loud)

Reply:
Hmm yes you are using `-var` option, me not and this could be an issue. So I may revert this part, as I saw that you have an Azure workaround in that case if I'm not wrong, no?

Reply:
Yes this one, and no we are creating only one RG (but with multiple VMs), so this workaround should work also.

I don't know exactly what method is better, and I was not able to find the good procedure for destroying on terraform: `plan ; apply ; destroy` or `plan ; apply ; plan -destroy ; apply`?